spark/src/main/scala/geotrellis/spark/knn/KNearestMethods.scala

```diff
@@ -0,0 +1,66 @@
+package geotrellis.spark.knn
+
+import scala.collection.immutable.Map
+
+import geotrellis.util.MethodExtensions
+import geotrellis.vector.{Extent, Point, Geometry, Feature}
+
+import org.apache.spark.rdd.RDD
+
+trait KNearestMethods[T] extends MethodExtensions[RDD[T]] {
+  def kNearest(x: Double, y: Double, k: Int)(f: T => Extent): Seq[T] = { KNearestRDD.kNearest[T](self, x, y, k)(f) }
+
+  def kNearest(p: (Double, Double), k: Int)(f: T => Extent): Seq[T] = { KNearestRDD.kNearest[T](self, p, k)(f) }
+
+  def kNearest(p: Point, k: Int)(f: T => Extent): Seq[T] = { KNearestRDD.kNearest[T](self, p, k)(f) }
```

Comment:
We don't need an f: T => Extent if it's a point. This is covered by the Geom methods.